### PR TITLE
Use jsonschema to validate messages

### PR DESCRIPTION
**Pull request by takluyver**
Instead of custom validation built on traitlets. JSONSchema is a format designed to express precisely this kind of thing.

I haven't thrown all messages into one giant schema, because I think that will make unhelpful errors. In nbformat we deal with that by catching the errors and doing some extra work to produce more useful errors, but in this case I think it's more straightforward to break the schema up and use the appropriate parts for the message type.

### CODE REVIEW

File: jupyter_kernel_test/msgspec_v5.py [hunk 1]
```diff
@@ -0,0 +1,336 @@
+"""Message schemas for message spec version 5"""
```
Comment (blink1073) [inline jupyter_kernel_test/msgspec_v5.py:1]: This is excellent!  I think this and the `.json` file it emits should be in `nbformat`.

Comment (takluyver) [reply jupyter_kernel_test/msgspec_v5.py:1]: I'd say `jupyter_client` would probably be the place for it - that's where the messaging protocol is defined. But I thought I'd put it here first, where there's a clear requirement for it.

Comment (blink1073) [reply jupyter_kernel_test/msgspec_v5.py:1]: Ah, right, agreed!

File: jupyter_kernel_test/msgspec_v5.py [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+# Stdin messages ---------------------------------------------
+
+schema_fragments["input_request"] = {"properties": {
+    "prompt": {"type": "string"},
+    "password": {"type": "number"},
+}}
+
+schema_fragments["input_reply"] = {"properties": {
+    "value": {"type": "string"},
+}}
```
Comment (rgbkrk) [inline jupyter_kernel_test/msgspec_v5.py:336]: This is all awesome, I'm so glad you've put all these schemas together.

**Comment by takluyver** [comment]
I'm going to merge this now. It's useful to have the schemas, and we can move them into `jupyter_client` at any point.

**Pull request merged**